toolchain/check/testdata/impl/min_prelude/forward_declarations.carbon

```diff
@@ -0,0 +1,2460 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/min_prelude/facet_types.carbon
+// EXTRA-ARGS: --custom-core
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/impl/min_prelude/forward_declarations.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/impl/min_prelude/forward_declarations.carbon
+
+// --- empty.carbon
+library "[[@TEST_NAME]]";
+
+interface I;
+impl {} as I;
+
+interface I {}
+impl {} as I {}
+
+// --- method.carbon
+library "[[@TEST_NAME]]";
+
+interface I;
+impl {} as I;
+
+interface I {
+  fn G();
+}
+impl {} as I {
+  fn G() {}
+}
+
+// --- combine.carbon
+library "[[@TEST_NAME]]";
+
+interface I;
+impl {} as I & I;
+
+interface I {}
+impl {} as I & I {}
+
+// --- associated_const.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let T:! type;
+}
+impl {} as I where .T = ();
+
+impl {} as I where .T = () {}
+
+// --- associated_const_compound_member_access.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let T:! type;
+}
+class C {}
+impl C as I where .T = ();
+
+let x: C.(I.T) = ();
+
+impl C as I where .T = () {}
+
+// --- associated_const_of_facet.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let T:! type;
+}
+class C {}
+impl C as I where .T = ();
+
+let x: (C as I).T = ();
+
+impl C as I where .T = () {}
+
+// --- fail_todo_unset_associated_const.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let T:! type;
+}
+class C {}
+impl C as I;
+
+// TODO: Should fail because the associated constant is not set,
```

Comment:
Looks like this TODO is now done? Should test just be named fail_..?

Reply:
Done.